app/src/src_nonfreenet/org/breezyweather/sources/nws/NwsService.kt

```diff
@@ -154,16 +173,60 @@ class NwsService @Inject constructor(
         location: Location,
         requestedFeatures: List<SourceFeature>,
     ): Observable<SecondaryWeatherWrapper> {
-        if (!isFeatureSupportedInSecondaryForLocation(location, SourceFeature.FEATURE_ALERT)) {
+        if (!isFeatureSupportedInSecondaryForLocation(location, SourceFeature.FEATURE_CURRENT) ||
+            !isFeatureSupportedInSecondaryForLocation(location, SourceFeature.FEATURE_ALERT)
+        ) {
             // TODO: return Observable.error(UnsupportedFeatureForLocationException())
             return Observable.error(SecondaryWeatherException())
         }
+        val station = location.parameters
+            .getOrElse(id) { null }?.getOrElse("station") { null }
+        if (station.isNullOrEmpty()) {
+            return Observable.error(InvalidLocationException())
+        }
 
-        return mApi.getActiveAlerts(
-            USER_AGENT,
-            "${location.latitude},${location.longitude}"
-        ).map {
-            convertSecondary(it)
+        val failedFeatures = mutableListOf<SourceFeature>()
+        val current = if (requestedFeatures.contains(SourceFeature.FEATURE_CURRENT)) {
+            mApi.getCurrent(
```

Comment:
Do that if `!stationId.isNullOrEmpty()` then in the else add to `failedFeatures`

Reply:
I have added the if-then-else.

But if we had caught this condition with `InvalidLocationException()` already at the start of the function, in theory we would never reach this failure here.